File: design_patterns/C++/Structural/bridge.cxx
#include <iostream>
#include <string>

class Theme {
  public:
	virtual std::string getColor() = 0;
};

class DarkTheme: public Theme {
  public:
	std::string getColor() {
		return "Dark Black";
	}
};

class LightTheme: public Theme {
  public:
	std::string getColor() {
		return "Light White";
	}
};

class AquaTheme: public Theme {
  public:
	std::string getColor() {
		return "Light Blue";
	}
};

class WebPage {
  public:
	virtual std::string getContent() = 0;
};

class About: public WebPage {
  protected:
	Theme* theme = nullptr;

  public:
	About(Theme* theme) {
		this->theme = theme;
	}

	std::string getContent() {
		return "About page in " + this->theme->getColor();
	}
};

class Careers: public WebPage {
  protected:
	Theme* theme = nullptr;

  public:
	Careers(Theme* theme) {
		this->theme = theme;
	}

	std::string getContent() {
		return "Careers page in " + this->theme->getColor();
	}
};

int main() {
	auto darkTheme = new DarkTheme();
	auto about = new About(darkTheme);
	auto careers = new Careers(darkTheme);
	std::cout << about->getContent() << std::endl;
	std::cout << careers->getContent() << std::endl;
	// ==========================
	// About page in Dark Black
	// Careers page in Dark Black
	// ==========================
	return 0;
}

File: design_patterns/Python/Structural/bridge.py
class WebPage:

    def get_content(self):
        raise NotImplementedError


class About(WebPage):

    def __init__(self, theme):
        self.theme = theme

    def get_content(self):
        return 'About page in ' + self.theme.get_color()


class Careers(WebPage):

    def __init__(self, theme):
        self.theme = theme

    def get_content(self):
        return 'Careers page in ' + self.theme.get_color()


class Theme:

    def get_color(self):
        pass


class DarkTheme(Theme):

    def get_color(self):
        return 'Dark Black'


class LightTheme(Theme):

    def get_color(self):
        return 'Off White'


class AquaTheme(Theme):

    def get_color(self):
        return 'Light Blue'


if __name__ == '__main__':

    dark_theme = DarkTheme()
    about = About(dark_theme)
    careers = Careers(dark_theme)
    assert about.get_content() == 'About page in Dark Black'
    assert careers.get_content() == 'Careers page in Dark Black'
